tests/unit/test_current_user.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from bokeh.document import Document
+from dataclasses import dataclass
+
+from mz_bokeh_package.utilities import CurrentUser
+from mz_bokeh_package.utilities import MZGraphQLClient
+
+API_KEY = "6fzQxEJL1YX9ldWUhHv5NPpiIcV7Gg8r03KwmeCqyRakj2nSAstDMTb4oB"
+SESSION_ID = "LME0EBfddYQCJqpZoO4ysYvynDXFLavC8hu3RzHW6e5e"
+USER_ID = "79e8e0f4-d2d6-4c16-b833-0a8986a6ce53"
+USER_NAME = "user_name"
+
+
+TESTS = [
+    # Test 1: session ID provided and is logged in CurrentUser._users. API key not requered.
+    {
+        "input": {
+            "session_active": True,
+            "logged_user": True,
+            "session_id": SESSION_ID,
+            "api_key": None,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 2: session ID provided and is not logged in CurrentUser._users. API key provided.
+    {
+        "input": {
+            "session_active": True,
+            "logged_user": False,
+            "session_id": SESSION_ID,
+            "api_key": API_KEY,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 3: session ID not provided and is not logged in CurrentUser._users. API key provided.
+    {
+        "input": {
+            "session_active": False,
+            "logged_user": False,
+            "session_id": None,
+            "api_key": API_KEY,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 4: session ID not provided and is not logged in CurrentUser._users. API key not provided.
+    {
+        "input": {
+            "session_active": False,
+            "logged_user": False,
+            "session_id": None,
+            "api_key": None,
+        },
+        "output": ValueError
+    },
+]
+
+
+@dataclass
+class SessionContext:
```

Comment:
This doesn't seem like a good idea to me... `SessionContext` is an existing class in Bokeh, and you should not create your own class to mimic it, that could cause all kinds of problems we don't expect now.

This is an example where I would adjust my code a bit to make it easier to test. If you had a private method in `CurrentUser`:
```
@staticmethod
def _get_session_id( -> str | None:
   return curdoc().session_context.id if curdoc().session_context else None
```
Then you would just need to mock this one method, instead of all the "hacks" that are needed now.